### PR TITLE
Update/boost plugin to use speed score package

### PR DESCRIPTION
**Pull request by IanRamosC**
Fixes 1201210512993648-as-1204651717398080

## Proposed changes:

* This PR udpates the Boost plugin to use the Boost Speed Score package added in #30914. I created this branched off of `add/jetpack-boost-speed-score-package` and I'm currently comparing this branch against trunk, but I'll rebase it as soon as #30914 is deployed.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1HpG7-k9o-p2

## Does this pull request change what data or activity we track or use?

Nope

## Testing instructions:

* Checkout this branch locally
* Run `jetpack build packages/boost-speed-score` and `jetpack install packages/boost-speed-score`
* Run `jetpack build plugins/boost` and `jetpack install plugins/boost`
* Alternatively, create a JN (or an Atomic Ephemeral) site with Jetpack Beta installed and with `update/boost-plugin-to-use-speed-score-package` as the selected branch
* Visit `/wp-admin/admin.php?page=jetpack-boost` and make sure Boost works as expected

### CODE REVIEW

File: projects/packages/boost-speed-score/.phpcsignore [hunk 1]
```diff
@@ -0,0 +1 @@
+src/class-example.php
```
Comment (robertsreberski) [inline projects/packages/boost-speed-score/.phpcsignore]: We don't have this file from what I see - maybe this rule is redundant?

**Review by robertsreberski — COMMENTED** [review]
One minor comment, not a blocker

**Review by robertsreberski — DISMISSED** [review]
The migration of code to standalone package looks good! For the future, these kind of changes could be broken down into several PRs instead - one that generates the package, second that moves the code to the package and updates references in Jetpack Boost library.

That way we would separate auto-generated code from the migration code making it easier to process.

**Comment by CodeyGuyDylan** [comment]
> The migration of code to standalone package looks good! For the future, these kind of changes could be broken down into several PRs instead - one that generates the package, second that moves the code to the package and updates references in Jetpack Boost library.

This is actually in separate PR's, this PR contains code from https://github.com/Automattic/jetpack/pull/30914 which has not been merged yet. Once that is merged and this is rebased, the PR should be much smaller 😄

**Comment by CodeyGuyDylan** [comment]
Had some trouble with the rebase but all should be good now. @robertsreberski if you wanted to look at this again, it should reflect more of what is actually changing now

**Review by thingalon — DISMISSED** [review]
Seems to work, and the code changes seem ok

**Review by robertsreberski — DISMISSED** [review]
Thanks @CodeyGuyDylan @IanRamosC , it looks clear now!

**Review by CodeyGuyDylan — APPROVED** [review]

**Pull request merged**